File: go/source.go
package main

import (
	"context"
	"fmt"

	"github.com/jackc/pgx/v4"
)

func main() {

	ctx := context.Background()
	connStr := "postgres://MATERIALIZE_USERNAME:APP_SPECIFIC_PASSWORD@MATERIALIZE_HOST:6875/materialize"

	conn, err := pgx.Connect(ctx, connStr)
	if err != nil {
		fmt.Println(err)
	} else {
		fmt.Println("Connected to Materialize!")
	}

	createSourceSQL := `CREATE SOURCE counter FROM LOAD GENERATOR COUNTER`

	_, err = conn.Exec(ctx, createSourceSQL)
	if err != nil {
		fmt.Println(err)
	}

	defer conn.Close(context.Background())
}

File: typescript/src/source.ts
// node 14+ cjs named exports not found in pg
import pkg from 'pg';
const { Client } = pkg;

const client = new Client({
    user: "MATERIALIZE_USERNAME",
    database: "materialize",
    password: "APP_SPECIFIC_PASSWORD",
    hostname: "MATERIALIZE_HOST",
    port: 6875
});

async function main() {
    try {
        await client.connect();
        const res = await client.query(
            `CREATE SOURCE counter FROM LOAD GENERATOR COUNTER`
            );
        console.log(res);
    } catch(e) {
        console.log("error: ", e);
    } finally {
        client.end();
    }
}

await main();